release.sh

```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
+
+# PLEASE DO NOT EDIT THIS FILE UNLESS YOU ARE ABSOLUTELY SURE A CHANGE NEEDS TO BE MADE
+
+# THE PURPOSE OF THIS FILE IS TO AUTOMATE THE RELEASE PROCESS FOR THE FLUX LSP.
+# ITS SOLE FUNCTION IS TO INCREMENT THE VERSION NUMBER IN PACKAGE.JSON, CREATE A TAG,
+# PUSH IT TO GITHUB, AND CREATE A RELEASE. THIS WILL TRIGGER A CI JOB THAT WILL
+# PUBLISH THE NEW RELEASE DIRECTLY TO NPM.
+
+# THERE SHOULD BE MINIMAL CHANGES INCLUDED IN THE COMMIT GENERATED BY THIS SCRIPT,
+# AND THEY SHOULD BE DONE IN THE MOST MECHANICAL WAY POSSIBLE.
+
+which hub &> /dev/null
+if [[ @? != 0 ]]; then
+	echo "Please install the hub command line tool before running this script."
+	echo "https://github.com/github/hub"
+	exit 1
+fi
+
+current_branch=$(git branch --show-current)
+if [[ $current_branch != "master" ]]; then
+	echo "You are not on the master branch. Aborting."
+	exit 1
+fi
+
+git_changes=$(git status -s | wc -l)
+if [[ $git_changes != 0 ]]; then
+	echo "You have modified the master branch."
```

Comment:
Yeah, see, here's a good example. Maybe some other script accidentally modified the files on disk, so the user didn't _mean_ to modify them. It seems like a really little thing, and maybe it is but, for me, it can be a canary in a coal mine of building things designed for humans.